### PR TITLE
authenticate token each time client is sending request

### PR DESCRIPTION
**Pull request by VeryEarly**
## Description

#11925 
#6585 

Added delegate function to `RenewingTokenCredential` when management client was instantiated.

This function will authenticate token (refresh if expired) each time `ProcessHttpRequestAsync` was called.

`ProcessHttpRequestAsync` will be executed before management client tries to send request (include polling for long running operation)
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/mgmtcommon/ClientRuntime.Azure/ClientRuntime.Azure/LRO/LROPollState.cs#L182

## Checklist

- [ ] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/Accounts/Authentication/Authentication/RenewingTokenCredential.cs [hunk 1]
```diff
@@ -17,22 +17,36 @@
...
+using System.Management.Automation.Language;
+using System;
+using System.ComponentModel;
 
 namespace Microsoft.Azure.Commands.Common.Authentication
 {
     public class RenewingTokenCredential : ServiceClientCredentials
     {
         private IAccessToken _token;
+        private readonly Func<IAccessToken> _f;
```
Comment (isra-fel) [inline src/Accounts/Authentication/Authentication/RenewingTokenCredential.cs:29]: How about `_refresh` or `_authenticate`

File: src/Accounts/Authentication/Authentication/RenewingTokenCredential.cs [hunk 2]
```diff
@@ -17,22 +17,36 @@
...
 
 
+        public RenewingTokenCredential(IAccessToken token, Func<IAccessToken> f)
```
Comment (isra-fel) [inline src/Accounts/Authentication/Authentication/RenewingTokenCredential.cs:32]: A default parameter can save you a few lines of code
```suggestion
        public RenewingTokenCredential(IAccessToken token, Func<IAccessToken> f = null)
```

I also suggest write some docs in comment

**Comment by isra-fel** [comment]
Thinking about the different login scenarios: interactive, service principal, managed service identity, they use different mechanism to refresh token. We need test them all.

File: src/Accounts/Authentication/Authentication/RenewingAccessTokenProvider.cs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+using System;
+using System.Collections.Generic;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using ClientRuntime = Microsoft.Rest;
+
+namespace Microsoft.Azure.Commands.Common.Authentication.Authentication
+{
+    internal class RenewingAccessTokenProvider : ClientRuntime.ITokenProvider
```
[the remark this answers is not in the document]
Comment (VeryEarly) [reply src/Accounts/Authentication/Authentication/RenewingAccessTokenProvider.cs:24]: updated as suggested

**Comment by VeryEarly** [comment]
> One minor comment. Also, what validation are we planning to do here?

1. I'll debug a long running operation see whether it was a new token (from context.account.accesstoken)

2. provide engineering build  to partner (who reported this issue), ask them to help verify (meanwhile I'll try to reproduce more than one hour long running operations in my local)

File: src/Accounts/Authentication/Authentication/RenewingAccessTokenProvider.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (markcowl) [reply src/Accounts/Authentication/Authentication/RenewingAccessTokenProvider.cs:24]: This isn't exactly what I had in mind.  We shouldn't need to make any change to the renewing access token if we implement I AccessToken.  Here is an implementation of IAccessToken as an example: https://github.com/Azure/azure-powershell/blob/master/src/Accounts/Authentication/Authentication/ServicePrincipalTokenProvider.cs#L162

In this case there would be no need to implement ClientRuntime.ITokenProvider in this class.  LMK if you would like to discuss.

The renewal is done during IAccessToken.AuthorizeRequest, so there should not be any need for changes to the RenewingTokenCredential class

**Pull request merged**